eng/scripts/download_targeted_packages.py

```diff
@@ -0,0 +1,155 @@
+# this script uses the package pipgrip to evaluate the targeted packages and return the full list of dependencies
+import argparse
+import os
+import shutil
+import re
+import json
+import bs4
+import requests
+from typing import List, Set, Tuple
+import subprocess
+
+def get_file(url: str, dest_folder: str) -> None:
+    """
+    Downloads a targeted URL to a destination folder.
+    """
+    file_name = url.split("/")[-1].replace(" ", "_")
+    file_name = file_name.split("#")[0]
+    file_path = os.path.join(dest_folder, file_name)
+
+    r = requests.get(url, stream=True)
+    if r.status_code == 200:
+        with open(file_path, "wb") as f:
+            for chunk in r.iter_content(chunk_size=1024 * 8):
+                if chunk:
+                    f.write(chunk)
+                    f.flush()
+                    os.fsync(f.fileno())
+    else:
+        print('Download failed with status code "{}" and error "{}".'.format(r.status_code, r.text))
+
+
+def get_pkg_tuple(package_specifier: str) -> Tuple[str, str]:
+    """
+    Takes a specifier of form "azure-core==1.20.0" and returns ("azure-core", "1.20.0")
+
+    If no specific version is specified, the latter part of the tuple will be None.
+    """
+    result = re.split(r"[><=]", package_specifier)
+
+    if len(result) == 3:
+        package_name = result[0]
+        package_version = result[2]
+
+    elif len(result) == 2:
+        package_name = result[0]
+        package_version = ""
+
+    return package_name, package_version
+
+
+def get_pkg_files(package_specifier: str) -> List[str]:
+    """
+    Given a package specifier "azure-core==1.20.0", find all wheels and source distributions that match that version.
+
+    Retrieves using the PyPI simple index. This will provide up-to-date CDN links to each download.
+    """
+    pkg_name, pkg_version = get_pkg_tuple(package_specifier)
+
+    if not pkg_version:
+        print("This script does not support a specifier-less target package. Please specify an exact version.")
+        return []
+
+    retrieval_string = "https://pypi.org/simple/{0}".format(pkg_name)
+    result = requests.get(retrieval_string)
+    soup = bs4.BeautifulSoup(result.text, "html.parser")
```

Comment:
Is scrapping the html best way to do this? 

For the package index I use they python api https://github.com/Azure/azure-sdk/blob/main/eng/scripts/Query-Azure-Packages.ps1#L114 to get the package list and then version info.